src/main/java/thecodex6824/thaumcraftfix/common/util/PechTradeHelper.java

```diff
@@ -0,0 +1,76 @@
+package thecodex6824.thaumcraftfix.common.util;
+
+import net.minecraft.init.Items;
+import net.minecraft.init.PotionTypes;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.potion.PotionUtils;
+import net.minecraftforge.registries.IForgeRegistry;
+import org.apache.logging.log4j.Logger;
+import thaumcraft.common.entities.monster.EntityPech;
+import thecodex6824.thaumcraftfix.ThaumcraftFix;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * A helper for adding and removing pech trades. Pech trades are initialized in {@link thaumcraft.common.config.ConfigEntities#initEntities(IForgeRegistry)}
+ * and stored in {@link EntityPech#tradeInventory}.
+ * <p>
+ * The trades are stored in a map as the format:  (int) PechVariant : { (int) TradeLevel, (ItemStack) TradeItem }
+ */
+@SuppressWarnings("rawtypes") //Thaumcraft uses rawtypes for all the trade values the errors are suppressed.
+public class PechTradeHelper {
+    public static void fixPechTrades() {
+        Logger logger = ThaumcraftFix.instance.getLogger();
+        try {
+            //Removing Pech trades with broken potion items caused by invalid metadata values.
+            removePechTrade(EnumPechType.MAGE, Ingredient.fromStacks(new ItemStack(Items.POTIONITEM, 1, 8193)));
+            removePechTrade(EnumPechType.MAGE, Ingredient.fromStacks(new ItemStack(Items.POTIONITEM, 1, 8261)));
+            //Re-adding the potion Pech trades.
+            addPechTrade(EnumPechType.MAGE, 2, PotionUtils.addPotionToItemStack(new ItemStack(Items.POTIONITEM), PotionTypes.REGENERATION));
```

Comment:
We should probably only add these trades if we successfully deleted the corresponding broken ones, in case some other mod also tries to fix the trades